backend/routes.py

```
@@ -35,50 +38,77 @@ def register_event_handlers():
+    x_hub_signature_256: str = Header(None)
+):
+    # Read body once to avoid stream exhaustion
+    body = await request.body()
+
+    # Verify GitHub signature
+    await verify_signature(body, x_hub_signature_256)
+
+    try:
+        payload = json.loads(body.decode())
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=400, detail="Invalid JSON payload")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)